python_modules/libraries/dagster-snowflake/dagster_snowflake/snowflake_io_manager.py

```diff
@@ -312,7 +317,21 @@ def _get_cleanup_statement(table_slice: TableSlice) -> str:
         )
         return query + _partition_where_clause(table_slice.partition_dimensions)
     else:
-        return f"DELETE FROM {table_slice.database}.{table_slice.schema}.{table_slice.table}"
+        return f"TRUNCATE {table_slice.database}.{table_slice.schema}.{table_slice.table}"
```

Comment:
I don't think we can support a TRUNCATE call here right now. There would be an issue if a user wrote an asset, materialized it, then changed the code to add a new column. When we re-materialize the asset, if we just truncate the table, the underlying snowflake table will still have the columns of the original materialization. When we try to load the new dataframe with the new column, we will get an error because the new dataframe column doesn't have a corresponding column in the snowflake table. 

Unfortunately, the library we are using to load dataframes (pandas.to_sql with the snowflake pandas writer) doesn't handle adding new columns. We have an issue open [here](https://github.com/dagster-io/dagster/issues/13098) to support adding new columns for partitions (where we have the same issue). Once we implement this, then we could switch to using TRUNCATE here as well

Reply:
My understanding is that `DELETE FROM` also preserves column names/schema?

Reply:
ah yeah, you are totally right,  i had bad info. Tested it and it does keep the columns. truncate seems like it should be good then. 

The issue with adding a new column to the dataframe in an asset exists in the current dagster version, so i'll update the gh issue. I'll give this another review pass, but it generally looks good